airbyte-cdk/python/airbyte_cdk/sources/file_based/README.md

```diff
@@ -1,3 +1,103 @@
+* `true_values`: As CSV does not explicitly define a value for positive boolean, the user can specify a set of case-sensitive strings that should be interpreted as true values.
+* `false_values`: As CSV does not explicitly define a value for negative boolean, the user can specify a set of case-sensitive strings that should be interpreted as false values.
+
+### JSONL
+[JSONL](https://jsonlines.org/) (or JSON Lines) is a format where each row is a JSON object. There are no configuration option for this format. For backward compatibility reasons, the JSONL parser currently supports multiline objects even though this is not part of the JSONL standard. Following some data gathering, we reserve the right to remove the support for this. Given that files have multiline JSON objects, performances will be slow. 
+
+### Parquet
+Parquet is a file format defined by [Apache](https://parquet.apache.org/). Configuration options are:
+* `decimal_as_float`: Whether to convert decimal fields to floats. There is a loss of precision when converting decimals to floats, so this is not recommended.
+
+## Schema
+
+Having a schema allows for the file-based CDK to take action when there is a discrepancy between a record and what are the expected types of the record fields.
+
+Schema can be either inferred or user provided.
+* If the user defines it a format using JSON types, inference will not apply. Input schemas are a key/value pair of strings describing column name and data type. Supported types are `["string", "number", "integer", "object", "array", "boolean", "null"]`. For example, `{"col1": "string", "col2": "boolean"}`.
+* If the user enables schemaless sync, schema will `{"data": "object"}` and therefore emitted records will look like `{"data": {"col1": val1, …}}`.
+* Else, the file-based CDK will infer the schema depending on the file type. Some file formats defined the schema as part of their metadata (like Parquet), some do on the record-level (like Avro) and some don't have any explicit typing (like JSON or CSV). Any file format that does not define their schema on a metadata level will require the file-based CDK to iterate to a number of records. There is a limit of bytes that will be consumed in order to infer the schema.
+
+### Validation Policies
+Users will be required to select one of 3 different options, in the event that records are encountered that don’t conform to the schema.
+
+* Skip nonconforming records: check each record to see if it conforms to the schema; keep a count of the number of records in each file that do and do not conform and emit a log message with these counts once we’re done reading the file. Do not emit records that don’t conform.
+* Emit all records: columns that don’t exist in the configured catalog may still be available in the destination’s raw tables, but this is undefined behavior.
```

Comment:
```suggestion
* Emit all records: emit all records, even if they do not conform to the user-provided or inferred schema. Columns that don’t exist in the configured catalog may still be available in the destination’s raw tables, but this is undefined behavior.
```

Reply:
nit: we should say that columns that don't exist in the configured catalog probably won't be available in the destination's table since that's the current behavior. (I agree it should still be considered undefined behavior)